### ISSUE TITLE
CRD.version is no longer generated at master

### ISSUE THREAD

**Issue by heoneal** [operator-framework/operator-sdk#2953]
## Bug Report

**What did you do?**
A clear and concise description of the steps you took (or insert a code snippet).
I ran the following code with the current master of opertor-framework.

```bash
#! /bin/bash
# Channel will be used in OLM to reference versions
export CHANNEL=alpha
# the csv version 
export CSVVER=0.0.1
# your operators name will be used
export OPERATOR_NAME=memcachedoperator
# deploy dir for gen csv cmd
export DEPLOY_DIR=deploy
# the location of the CRDs for gen csv
export CRD_DIR=deploy/crds
# the location of the uniuq TYPE for this Operator for gen csv
export API_DIR=pkg/apis

operator-sdk version;
operator-sdk new ${OPERATOR_NAME} --repo=github.com/example-inc/${OPERATOR_NAME};
cd ${OPERATOR_NAME};
operator-sdk add api --api-version=cache.example.com/v1alpha1 --kind=Memcached;
operator-sdk add controller --api-version=cache.example.com/v1alpha1 --kind=Memcached;

operator-sdk generate k8s;
operator-sdk generate crds;

```

**What did you expect to see?**
A clear and concise description of what you expected to happen (or insert a code snippet).

I expected to see the spec.version field of the CRD filled with v1alpha1

**What did you see instead? Under which circumstances?**
A clear and concise description of what you expected to happen (or insert a code snippet).

The CRD was produced without the spec.version field.

**Environment**
* operator-sdk version:

operator-sdk version: "v0.17.0-54-g5279f2fc", commit: "5279f2fcb779f4f39e23e66b1c8d0e027b181f7c", kubernetes version: "v1.18.2", go version: "go1.13 linux/amd64"

* go version:

go version go1.13 linux/amd64

* Kubernetes version information:

% kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.6", GitCommit:"96fac5cd13a5dc064f7d9f4f23030a6aeface6cc", GitTreeState:"clean", BuildDate:"2019-08-19T11:13:49Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:07:57Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}

* Kubernetes cluster kind: 
kubernetes in docker-desktop

* Are you writing your operator in ansible, helm, or go?
Go

**Possible Solution**
Revert changes in master that broke this in the v0.17.0

**Additional context**
The spec.version is properly generated in the CRD

**Comment by camilamacedo86** [comment]
Hi @heoneal,

From now, the tool is generating CRDs with the version `v1` instead of `apiextensions.k8s.io/v1beta1`.See the [Cnagelog](https://github.com/operator-framework/operator-sdk/blob/master/CHANGELOG.md#added) 

> The --crd-version flag was added to the new, add api, add crd, and generate crds commands so that users can opt-in to v1 CRDs. (#2684)

The `spec.version` was deprecated and remove in the `v1`. More info: [CRD k8s doc](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/). Now it is `versions` and is a list. See: 

```yaml
...
spec:
  # list of versions supported by this CustomResourceDefinition
  versions:
    - name: v1
    ...
```

If you wish still using the previous version of the k8s api, where the version field exists you can use the flag `--crd-version`. If you run the  command with the flag `--help` you are able to see: 

> --crd-version string          CRD version to generate (default "v1")

However, note that the previous version of CRD (`apiextensions.k8s.io/v1beta1`) is deprecated and will no longer be supported after 1.19 k8s version. 

Closing it as sorted out. However, please feel free to re-open if you see that it would be required.

**Comment by joelanford** [comment]
TL;DR: use `operator generate crds --crd-version=v1beta1`

Here's the PR where this change was made, for reference: #2874

**Comment by heoneal** [comment]
Thank you for the detail @camilamacedo86 I had missed the changelog and that makes a lot more sense.
